src/pal/inc/unixasmmacrosarm.inc

```diff
@@ -30,64 +41,43 @@ C_FUNC(\Name\()_End):
 .endm
 
 .macro push_nonvol_reg Register
-        push {\Register}
-        .cfi_adjust_cfa_offset 4
-        .cfi_rel_offset \Register, 0
+        push \Register
+        .save \Register
 .endm
 
 .macro pop_nonvol_reg Register
-        pop {\Register}
-        .cfi_adjust_cfa_offset -4
-        .cfi_restore \Register
-.endm
-
-.macro alloc_stack Size
-        sub sp, sp, \Size
-        .cfi_adjust_cfa_offset \Size
+        pop \Register
```

Comment:
I wonder if the .pad annotation should not be there. A good way to see if the unwinding information is correct is to step through e.g. the CallDescrWorkerInternal assembly method and see if the debugger "bt" commands shows the same backtrace at every instruction (well, it is enough to verify it around locations that push / pop registers or modify sp)